source/slang/slang-ir-entry-point-uniforms.cpp

```diff
@@ -415,10 +425,29 @@ struct MoveEntryPointUniformParametersToGlobalScope
 };
 
 void moveEntryPointUniformParamsToGlobalScope(
-    IRModule*   module)
+    IRModule*   module,
+    CodeGenTarget target)
 {
     MoveEntryPointUniformParametersToGlobalScope context;
+    
     context.module = module;
+    context.codeGenTarget = target;
+    context.targetNeedsConstantBuffer = true;
+
+    // Check if this target needs constant buffer wrapping
+    switch (target)
+    {
+        case CodeGenTarget::CPPSource:
+        case CodeGenTarget::CSource:
+        case CodeGenTarget::Executable:
+        case CodeGenTarget::SharedLibrary:
+        {
+            context.targetNeedsConstantBuffer = false;
```

Comment:
Looking at this logic.... I'm not sure I agree with this change. AFAICT it is *correct* for the CPU target to be wrapping the entry-point uniforms in a constant buffer, and indeed in the code you generate you make sure that the entry-point uniforms are indirected (an `_S1*` instead of an `_S1`), which is exactly what wrapping them in a constant buffer here was supposed to achieve.

Certainly at this point the *layout* has already been computed as if the entry-point uniforms would be wrapped, so ignoring that here is also ignore the layout that has already been set up.

I *think* you should be able to just revert all these edits to this file and make things work just fine, but maybe I'm missing something.